internal/jimm/juju/model_poller.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2025 Canonical.
+
+package juju
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/juju/juju/rpc/params"
+	jujuparams "github.com/juju/juju/rpc/params"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/servermon"
+)
+
+// PollModels loops over models, contacting the respective controller
+// and checking, based on the model's migration mode, if the model exists.
+// If the model exists in JIMM's database, but not on the controller,
+// it is deleted from JIMM's database.
+func (j *JujuManager) PollModels(ctx context.Context) (err error) {
+	const op = errors.Op("jimm.PollModels")
+	zapctx.Info(ctx, string(op))
+	durationObserver := servermon.DurationObserver(servermon.JimmMethodsDurationHistogram, string(op))
+	defer durationObserver()
+
+	// Step 1: Group models by controller
+	controllerModels := make(map[string][]*dbmodel.Model)
+	err = j.Database.ForEachModel(ctx, func(m *dbmodel.Model) error {
+		key := m.Controller.UUID
+		controllerModels[key] = append(controllerModels[key], m)
+		return nil
+	})
+	if err != nil {
+		return errors.E(op, err)
+	}
+
+	// Step 2: Loop over controllers and process their models
+	// This way we only dial each controller once.
+	for controllerUUID, models := range controllerModels {
+		if len(models) == 0 {
+			continue
+		}
+		api, err := j.dialController(ctx, &models[0].Controller)
+		if err != nil {
+			zapctx.Error(ctx, "cannot dial controller", zap.String("controller", controllerUUID), zap.Error(err))
+			continue
+		}
+		// Depending the model's migration mode, we either:
+		// - Check if the model exists (MigrationModeNone)
+		// - Check if the model has completed internal migration (MigrationModeMigrateInternal)
+		// - Do nothing if the model is in any other migration mode (MigrationModeImporting, MigrationModeExporting)
+		for _, m := range models {
+			ctx := zapctx.WithFields(ctx,
+				zap.String("modelOwner", m.OwnerIdentityName),
```

Comment:
We haven't got much consistency on label keys, I tend to use hyphens/snake and this is camel. What do you think about sticking with "model-owner" etc for the labels?